nbd/src/main.cpp

```diff
@@ -95,6 +95,84 @@ static void Usage() {
         << std::endl;
 }
 
+// use for record image and device info to auto map when boot
+static void AddRecord() {
+    std::fstream fs;
+    // check if exist
+    fs.open(CURVETAB_PATH, std::fstream::in);
+    if (!fs.is_open()) {
+        std::cerr << "curve-nbd: open curvetab file failed.";
+    }
+
+    std::string lineString;
+    while (getline(fs, lineString)) {
+        if (!lineString.empty() && lineString[0] != '#') {
+            std::vector<std::string> lineArray;
+            std::stringstream ss(lineString);
+            std::string str;
+            while (getline(ss, str, '\t')) {
+                lineArray.push_back(str);
+            }
+            if (lineArray.size() != 2 && lineArray.size() != 3) {
```

Comment:
Under what circumstances lineArray.size() == 2

Reply:
> Under what circumstances lineArray.size() == 2

if you only use curve-nbd map, will only image and device write in curvetab like: cbd:pool//test_test_    /dev/nbd0
then if you mkfs on the device, you need record the mount point in the curvetab manually like fstab.
reason for not using fstab is fstab run before rc or system.d at reboot.